docs/spec.md

```diff
@@ -541,6 +541,55 @@ bytes is implementation-defined. String literals have type `string`.
 
 ## Ops
 
+The following helper functions are defined to simplify the specification of the
+semantics of StableHLO operations with quantized types.
+
+```python
+##  operand: tensor with quantized type.
+##  unary_computation: unary function of type: (floating-point) ->
+##      floating-point.
+##  out_storage_min, out_storage_max, out_scale, and out_zero_point: integer
+##      constants
+def unary_computation_with_quantized_types(operand, unary_computation,
+    out_storage_min, out_storage_max, out_scale, out_zero_point):
+  float_result = unary_computation((operand - zero_point(operand)) *
+          scale(operand))
+  rounded_result = round_nearest_even(float_result / out_scale)
+  result = clamp(out_storage_min, rounded_result + out_zero_point,
+          out_storage_max)
+  return result
+
+##  lhs and rhs: tensors with quantized types.
+##  binary_computaion: binary function of type: (floating-point,
+##      floating-point) -> floating-point.
+##  out_storage_min, out_storage_max, out_scale, and out_zero_point: integer
+##      constants
+def binary_computation_with_quantized_types(lhs, rhs, binary_computaion,
+    out_storage_min, out_storage_max, out_scale, out_zero_point):
+  float_result = binary_computaion((lhs - zero_point(lhs)) * scale(lhs),
+          (rhs - zero_point(rhs)) * scale(rhs))
+  rounded_result = round_nearest_even(float_result / out_scale)
+  result = clamp(out_storage_min, rounded_result + out_zero_point,
+          out_storage_max)
+  return result
+
+##  operand1, operand2 and operand3: tensors with quantized types.
+##  ternary_computaion: ternary function of type: (floating-point,
+##      floating-point, floating-point) -> floating-point.
+##  out_storage_min, out_storage_max, out_scale, and out_zero_point: integer constants
+def ternary_computation_with_quantized_types(operand1, operand2, operand3,
+        ternary_computaion, out_storage_min, out_storage_max, out_scale,
+        out_zero_point):
+  float_result = ternary_computaion(
+          (operand1 - zero_point(operand1)) * scale(operand1),
+          (operand2 - zero_point(operand2)) * scale(operand2),
+          (operand3 - zero_point(operand3)) * scale(operand3))
+  rounded_result = round_nearest_even(float_result / out_scale)
+  result = clamp(out_storage_min, rounded_result + out_zero_point,
+                 out_storage_max)
+  return result
+```
```

Comment:
Might be a dumb question. Why not use `quantize` and `dequantize` function directly inside `dequantize_op_quantize`?

Reply:
Good point! Since we don't yet have specs for UniformQuantizeOp and UniformDequantizeOp, we don't yet have the corresponding `uniform_quantize` and `uniform_dequantize` functions available to use in our formulas. But once we do, I agree we should switch to them.

Reply:
makes sense to me.